py-polars/polars/utils/_construction.py

```diff
@@ -418,7 +418,14 @@ def sequence_to_pyseries(
         pyseries = _construct_series_with_fallbacks(
             constructor, name, values, dtype, strict=strict
         )
-        if dtype in (Date, Datetime, Duration, Time, Categorical, Boolean):
+        if dtype in (
+            Date,
+            Datetime,
+            Duration,
+            Time,
+            Boolean,
+            Categorical,
+        ) or isinstance(dtype, Categorical):
```

Comment:
This change won't be necessary if the Python class is implemented properly (see my other comment).